R/model.R

```diff
@@ -47,3 +47,44 @@ run_simulation <- function(pop, parameters = NULL, max_age = 100) {
   output
 
 }
+
+#' @title Run the simulation with repetitions
+#'
+#' @param repetitions n times to run the simulation
+#' @param overrides a named list of parameters to use instead of defaults
+#' @param parallel execute runs in parallel
+#' @return data frame for runs
+#' @export
+run_multiple_models <- function(
+  repetitions,
+  overrides = list(),
+  parallel = FALSE
+) {
+
+
+  if (parallel) {
+    fapply <- parallel::mclapply
+  } else {
+    fapply <- lapply
+  }
+
+  sum <- 0
+
+  dfs <- fapply(
+
+    seq(repetitions),
+    function(repetition) {
+      sum <- sum + 1
```

Comment:
this does not alter the sum on line 71, for that use `<<-`, although that will not work in parallel.

However, this could be - and I think should be - replaced by indexing by `repetition` which appears to be the intended behaviour. As tests are passing, we need a test here to make sure that these overrides make it through; a mock-based test here will likely be best

Reply:
It doesn't work using "repetition" for some reason so I used the old way renaming sum to counter. Mock test to be done